receiver/sqlserverreceiver/factory.go

```diff
@@ -256,10 +248,52 @@ func setupLogsScrapers(params receiver.Settings, cfg *Config) ([]scraperhelper.C
 }
 
 func isDatabaseIOQueryEnabled(metrics *metadata.MetricsConfig) bool {
+	if metrics == nil {
+		return false
+	}
+
 	if metrics.SqlserverDatabaseLatency.Enabled ||
 		metrics.SqlserverDatabaseOperations.Enabled ||
 		metrics.SqlserverDatabaseIo.Enabled {
 		return true
 	}
 	return false
 }
+
+func isPerfCounterQueryEnabled(metrics *metadata.MetricsConfig) bool {
+	if metrics == nil {
+		return false
+	}
+
+	if metrics.SqlserverBatchRequestRate.Enabled ||
```

Comment:
Nit: This could just be a return of the long `||` chain.

Reply:
Done